common/fee/models.go

```diff
@@ -0,0 +1,183 @@
+	return max(new(big.Int).Add(originalFeePrice, feeBumpUnits), addPercentage(originalFeePrice, feeBumpPercent))
+}
+
+func maxBumpedFee(lggr logger.SugaredLogger, currentFeePrice, bumpedFeePrice, maxFeePrice *big.Int, feeType string) *big.Int {
+	if currentFeePrice != nil {
+		if currentFeePrice.Cmp(maxFeePrice) > 0 {
+			// Shouldn't happen because the estimator should not be allowed to
+			// estimate a higher fee than the maximum allowed
+			lggr.AssumptionViolationf("Ignoring current %s of %s that would exceed max %s of %s", feeType, currentFeePrice.String(), feeType, maxFeePrice.String())
+		} else if bumpedFeePrice.Cmp(currentFeePrice) < 0 {
+			// If the current fee price is higher than the old price bumped, use that instead
+			bumpedFeePrice = currentFeePrice
+		}
+	}
+	return bumpedFeePrice
+}
+
+// CalculateFee computes the fee price and chain specific fee limit for a transaction.
+func CalculateFee(cfg feetypes.FixedPriceEstimatorConfig, bumpCfg feetypes.BumpConfig, feeLimit uint32, maxFeePrice *big.Int) (feePrice *big.Int, chainSpecificFeeLimit uint32, err error) {
+	feePrice = cfg.PriceDefault()
+	feePrice, chainSpecificFeeLimit = CapFeePrice(feePrice, maxFeePrice, bumpCfg.PriceMax(), feeLimit, bumpCfg.LimitMultiplier())
+	return
+}
+
+func GetDynamicFee(cfg feetypes.FixedPriceEstimatorConfig, originalFeeLimit uint32, maxFeePrice *big.Int) (feeCap, tipCap *big.Int, chainSpecificFeeLimit uint32, err error) {
+	tipCap = cfg.TipCapDefault()
+
+	if tipCap == nil {
+		return big.NewInt(0), big.NewInt(0), 0, errors.New("cannot calculate dynamic fee: FeeTipCapDefault was not set")
+	}
+
+	chainSpecificFeeLimit = ApplyMultiplier(originalFeeLimit, cfg.LimitMultiplier())
+	feeCap = GetFeeCap(cfg, originalFeeLimit, maxFeePrice)
+
+	return feeCap, tipCap, chainSpecificFeeLimit, nil
+
+}
+
+func GetFeeCap(cfg feetypes.FixedPriceEstimatorConfig, originalFeeLimit uint32, maxFeePrice *big.Int) (feeCap *big.Int) {
+	if cfg.BumpThreshold() == 0 {
+		// Fee bumping is disabled, just use the max fee cap
+		feeCap = getMaxFeePrice(maxFeePrice, cfg.PriceMax())
+	} else {
+		// Need to leave headroom for bumping so we fallback to the default value here
+		feeCap = cfg.FeeCapDefault()
+	}
+	return feeCap
+}
+
+// BumpDynamicFeeOnly bumps the tip cap and max fee price if necessary
+func CalculateBumpDynamicFee(cfg feetypes.BumpConfig, feeCapBufferBlocks uint16, lggr logger.SugaredLogger, currentTipCap, currentBaseFee *big.Int, originalFeeCap, originalTipCap *big.Int, originalFeeLimit uint32, maxFeePrice *big.Int) (bumpedFeeCap, bumpedTipCap *big.Int, chainSpecificFeeLimit uint32, err error) {
+	bumpedFeeCap, bumpedTipCap, err = bumpDynamicFee(cfg, feeCapBufferBlocks, lggr, currentTipCap, currentBaseFee, originalFeeCap, originalTipCap, maxFeePrice)
+	if err != nil {
+		return bumpedFeeCap, bumpedTipCap, 0, err
+	}
+	chainSpecificFeeLimit = ApplyMultiplier(originalFeeLimit, cfg.LimitMultiplier())
+	return
```

Comment:
This func looks like it is doing two separate things - can we separate these? Or maybe just inline them at that point? The callers already have to handle the error anyways, so I think it will be simpler to read.
Old:
```go
bumpedFeeCap, bumpedTipCap, chainSpecificGasLimit, err :=
		commonfee.CalculateBumpDynamicFee(f.config, f.bhConfig.EIP1559FeeCapBufferBlocks(),
			f.lggr, f.config.TipCapDefault(),
			nil, originalFee.FeeCap.ToInt(), originalFee.TipCap.ToInt(),
			originalGasLimit, maxGasPriceWei.ToInt())
	if err != nil {
		return DynamicFee{}, 0, err
	}
```
New:
```go
bumpedFeeCap, bumpedTipCap, err :=
		commonfee.BumpDynamicFee(f.config, f.bhConfig.EIP1559FeeCapBufferBlocks(),
			f.lggr, f.config.TipCapDefault(), nil, originalFee.FeeCap.ToInt(), 
			originalFee.TipCap.ToInt(), maxGasPriceWei.ToInt())
	if err != nil {
		return DynamicFee{}, 0, err
	}
	chainSpecificFeeLimit = commonfee.ApplyMultiplier(originalGasLimit, f.config.LimitMultiplier())
```

Reply:
ahh got it. `chainSpecificFeeLimit = ApplyMultiplier(originalFeeLimit, cfg.LimitMultiplier())` Yeah this part seems to be pretty confusing. i think that would be a good idea to seperate them as well.

Reply:
when i was reading the code, the part seemed to confuse me a little. Will extract out chain specific gas limit. Thanks for the note Jordan!